doc/src/vtr/server_mode/index.rst

```diff
@@ -0,0 +1,203 @@
+.. _server_mode:
+
+Server Mode
+================
+
+VTR provides the ability to run in server mode using the following command-line arguments.
+
+.. code-block:: none
+
+  --server --port_num 60555
+
+Server mode may only accept a single client application connection and respond to two types of requests: **get critical path report** and **highlight selected critical path elements**.
+
+Communication telegram
+-------------------------
+
+Telegram consists of two parts: a fixed-size **telegram header** and a variable-size **telegram body**.
+The telegram header contains helper information required to properly extract the telegram body sequence from the data flow.
+
+.. _fig_comm_telegram_structure:
+
+.. figure:: comm_telegram_structure.*
+
+    Communication telegram structure.
+
+    
+.. _fig_comm_telegram_body_structure:
+
+.. figure:: comm_telegram_body_structure.*
+
+    Communication telegram body structure.
+
+    **telegram body** is flat **JSON** structure.
+
+    **CMD** could have following integer values:
+
+    - 0 - command id for **get critical path**
+    - 1 - command id for **highlight selected path elements**
+
+    JOB_ID - is unique id for a task.
+
+    .. note:: The telegram body itself could be compressed with zlib to minimize the amount of data transferred over the socket.
+      This compression is applied to the response of the 'get critical path report' request.
+      The compressor ID byte in the telegram header signals whether the telegram body is compressed.
```

Comment:
Say what value means what.  (1 means compressed, 0 means not compressed?)

Reply:
I'd also describe the checksum (purpose, function to call to compute it).

Reply:
solved https://github.com/w0lek/vtr-verilog-to-routing/pull/11/commits/4f28a00e7c70f6d37d06b7106dab699da7bbc4c5
(will be updated later)